src/mono/wasm/features.md

```diff
@@ -2,36 +2,125 @@
+
+See also [CSP on MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP)
+
+## ICU
+
+_TODO_
+
+## Timezones
+
+Browsers don't offer API for working with time zone database and so we have to bring the time zone data as part of the application.
+
+If your application doesn't need to work with TZ, you can reduce download size by `<InvariantTimezone>true</InvariantTimezone>`.
+This requires that you have [wasm-tools workload](#wasm-tools-workload) installed.
+
 # Shell environments - NodeJS & V8
 We pass most of the unit tests with NodeJS v 14 but it's not fully supported target platform. We would like to hear about community use-cases.
-We also use v8 engine version 11 or higher to run some of the tests. The engine is lacking many APIs and features.
\ No newline at end of file
+
+We also use v8 engine version 11 or higher to run some of the tests. The engine is lacking many APIs and features.
+
+# Mobile phones
+
+Recent mobile phones have browser integration which could be upgraded separately from the operating system.
+
+Also note that all browsers on iOS are just wrapper for Safari engine and so it's limited by Safari features and version, rather than wrapper version and brand.
+
+Mobile phones usually have limited resources. Memory and download speed are major concern.
+
+## Resources consumed on the target device
+
+dotnet is complex and large application, it consists of
+- dotnet runtime, including garbage collector, IL interpreter and browser specific JIT
+- dotnet base class library
+- emulation layer which is bringing missing features of the OS, which the browser doesn't provide. Like timezone database or ICU.
+- integration with the browser JavaScript APIs, for example HTTP and WebSocket client
+- application code
+
+All of the mentioned code and data need to be downloaded and loaded into the browser memory during the dotnet startup sequence.
+Browser itself will run JIT compilation of the WASM and JS code, which consumes memory and CPU cycles too.
+
+## WASM linear memory
+
+You can override initial size of the WASM linear memory by `<EmccInitialHeapSize>16777216</EmccInitialHeapSize>`.
+This requires that you have [wasm-tools workload](#wasm-tools-workload) installed.
+
+## JITerpreter
+
+Is browser specific JIT compiler which optimizes small fragments of code which are otherwise interpreted by dotnet mono interpreter. It's enabled by default.
+
+## AOT
+
+You can enable Ahead Of Time compilation by `<RunAOTCompilation>true</RunAOTCompilation>`.
+This requires that you have [wasm-tools workload](#wasm-tools-workload) installed.
+
+It will compile managed code as native WASM instructions and include them in the `dotnet.native.wasm` file.
+AOT compiled code is running faster but it will significantly increase size of the file and the download time.
+
+## IL trimming
+
+You can trim size of the managed code in the assemblies by `<PublishTrimmed>true</PublishTrimmed>`.
```

Comment:
It's probably worth explaining in more detail what IL trimming does and why you would want to use it, like
`Trimming will remove unused code from your application, which reduces download time and memory usage. When AOT compilation is in use, trimming will also reduce the amount of time spent to compile the application.`

Reply:
We may also want to warn that `Some applications will break if trimming is used without further configuration due to the trimmer not knowing which code is used` and link to the documentation on trimming so users can configure it.